pepdbagent/base_connection.py

```diff
@@ -0,0 +1,130 @@
+from urllib.parse import urlparse
+import logging
+import psycopg2
+from .const import *
+from .exceptions import SchemaError
+
+_LOGGER = logging.getLogger("pepdbagent")
+
+
+class BaseConnection:
+    """
+    A class with base methods, that are used in several classes. e.g. fetch_one or fetch_all
+    """
+
+    def __init__(
+        self,
+        host="localhost",
+        port=5432,
+        database="pep-db",
+        user=None,
+        password=None,
+        dsn=None,
+    ):
+        """
+        Initialize connection to the pep_db database. You can use The basic connection parameters
+        or libpq connection string.
+        :param host: database server address e.g., localhost or an IP address.
+        :param port: the port number that defaults to 5432 if it is not provided.
+        :param database: the name of the database that you want to connect.
+        :param user: the username used to authenticate.
+        :param password: password used to authenticate.
+        :param dsn: libpq connection string using the dsn parameter
+        (e.g. "localhost://username:password@pdp_db:5432")
+        """
+
+        _LOGGER.info(f"Initializing connection to {database}...")
+
+        if dsn is not None:
+            self.pg_connection = psycopg2.connect(dsn)
+            self.db_name = urlparse(dsn).path[1:]
```

Comment:
It's not immediately clear what this does - maybe move to a helper function with a descriptive name?

Reply:
all class ?

Reply:
I was just saying makings something like:

```python
self._extract_db_name(dsn: str) -> str:
    return urlparse(dsn).path[1:]
```